pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -1376,33 +1371,35 @@ func pruneBundle(tx *gorm.DB, trustDomainID string, expiry time.Time, log logrus
 	return changed, nil
 }
 
-func taintX509CA(tx *gorm.DB, trustDomainID string, publicKeyToTaint crypto.PublicKey) error {
+func taintX509CA(tx *gorm.DB, trustDomainID string, subjectKeyIDToTaint string) error {
 	bundle, err := getBundle(tx, trustDomainID)
 	if err != nil {
 		return err
 	}
 
-	for _, eachTaintedKey := range bundle.X509TaintedKeys {
-		taintedKey, err := x509.ParsePKIXPublicKey(eachTaintedKey.PublicKey)
+	found := false
+	for _, eachRootCA := range bundle.RootCas {
+		x509CA, err := x509.ParseCertificate(eachRootCA.DerBytes)
 		if err != nil {
-			return status.Errorf(codes.Internal, "failed to parse tainted Key: %v", err)
+			return status.Errorf(codes.Internal, "failed to parse rootCA: %v", err)
 		}
 
-		ok, err := cryptoutil.PublicKeyEqual(taintedKey, publicKeyToTaint)
-		if err != nil {
-			return status.Errorf(codes.Internal, "failed to compare public key: %v", err)
+		caSubjectKeyID := x509util.SubjectKeyIDToString(x509CA.SubjectKeyId)
+		if subjectKeyIDToTaint != caSubjectKeyID {
+			continue
 		}
-		if ok {
+
+		if eachRootCA.TaintedKey {
 			return status.Errorf(codes.InvalidArgument, "root CA is already tainted")
 		}
-	}
 
-	pKey, err := x509.MarshalPKIXPublicKey(publicKeyToTaint)
-	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "failed to marshal public key to taint: %v", err)
+		found = true
+		eachRootCA.TaintedKey = true
 	}
 
-	bundle.X509TaintedKeys = append(bundle.X509TaintedKeys, &common.X509TaintedKey{PublicKey: pKey})
+	if !found {
+		return status.Errorf(codes.NotFound, "no ca found with provided subject key ID")
+	}
 
 	_, err = updateBundle(tx, bundle, nil)
```

Comment:
I think that we should increment bundle.SequenceNumber before calling updateBundle, otherwise SequenceNumber will remain the same.